packages/share_plus/share_plus_windows/lib/share_plus_windows.dart

```diff
@@ -42,15 +46,38 @@ class ShareWindows extends SharePlatform {
   }
 
   /// Share files.
+  /// [paths] pass only 1 file for Windows platform
+  /// [text] [subject] share text NOT supported for Windows platform
   @override
   Future<void> shareFiles(
     List<String> paths, {
     List<String>? mimeTypes,
     String? subject,
     String? text,
     Rect? sharePositionOrigin,
-  }) {
-    throw UnimplementedError(
-        'shareFiles() has not been implemented on Windows.');
+    ShareWithApp? appName,
+  }) async {
+    try {
+      if (appName == null) {
+        appName == ShareWithApp.BY_DEFAULT_APP;
+      }
+      await Process.run(
+        ShareWithAppCmd[appName]!,
+        [paths.first.toString()],
+        runInShell: true,
+      );
+    } catch (e) {
+      throw Exception('Unable to share on windows');
+    }
   }
+
+  Map<ShareWithApp, String> ShareWithAppCmd = {
```

Comment:
I see that limiting it to these choices does make it more convenient to use. On the other hand, it also limits the functionality tremendously and does seem a little redundant to me. Maybe provide defaults (like these) but allow the user to supply any string for the invocation.

If we do enable that, however, we drift further away from the mobile API and add a "feature" that seems a little redundant if the whole thing is basically:

```dart
await Process.run(
        ProgramToInvoke,
        fileToShare,
        runInShell: true,
);
```

share_plus to me focuses on the _native_ implementations - yes there is a native share feature, especially in Windows 11: #333 - so while this seems really useful to many, I don't see it fitting in with share_plus.

Reply:
I can implement it with `shareFilesWithResult` which return `ShareResultStatus.unavailable` as I didn't found if command returns any result. 
  
https://github.com/fluttercommunity/plus_plugins/issues/333#issuecomment-1085777282 , I can set to `ShareWithAppWindows.BY_DEFAULT_APP` because desktop app may have to open specific folder and files.

Should I proceed with this?